mergin/client.py

```diff
@@ -455,9 +453,19 @@ def apply_pull_changes(self, changes, temp_dir):
                             if os.path.exists(f'{dest}-shm'):
                                 os.remove(f'{dest}-shm')
                     else:
-                        # just use server version of file to update both project file and its basefile
-                        shutil.copy(src, dest)
-                        shutil.copy(src, basefile)
+                        # The local file is not modified -> no rebase needed.
+                        # We just apply the diff between our copy and server to both the local copy and its basefile
+                        try:
+                            server_diff = self.fpath(f'{path}-server_diff', temp_dir)  # diff between server file and local basefile
+                            self.geodiff.create_changeset(basefile, src, server_diff)
```

Comment:
it seems now that this is common for both cases of un/modified file, maybe move it up?

Reply:
If it is OK I would do that later - also with the cleanup of the code above (copying of files, removal of WAL etc)